Schemagen source option

johnpoth opened the pull request:
Fixes #2 , tested on [Apache Camel](https://github.com/apache/camel), thanks !

davidmoten commented on jax-maven-plugin/src/main/java/com/github/davidmoten/jaxws/HasSourcesOption.java:
```diff
@@ -0,0 +1,9 @@
+
+import java.util.List;
+
+interface HasSourcesOption {
```
This interface is unnecessary because schemagen is the only goal that has input source. Just do a check if `this` in `BaseMojo.java` is an instanceof `SchemagenMojo`.

johnpoth replied:
Done

johnpoth commented on jax-maven-plugin/src/main/java/com/github/davidmoten/jaxws/WsImportMojo.java:
```diff
@@ -1,8 +1,9 @@
+import org.apache.maven.plugins.annotations.ResolutionScope;
 
-@Mojo(name = "wsimport")
+@Mojo(name = "wsimport", requiresDependencyResolution = ResolutionScope.COMPILE)
```
Without specifying the ResolutionScope, the plugin wouldn't resolve the `required` scope dependencies leading to ClassNotFound exceptions

johnpoth commented on jax-maven-plugin/src/main/java/com/github/davidmoten/jaxws/BaseMojo.java:
```diff
@@ -245,9 +253,45 @@ private void addSources(Log log) {
+                    try {
+                        Files.walkFileTree(path, visitor);
+                    } catch (IOException e) {
+                        throw new RuntimeException("Error visiting sources", e);
```
Done

johnpoth commented:
@davidmoten done! Added a quick line in the features list. Hope that's what you had in mind.

I also added another commit which sets the working directory of the process to that of the executing Maven module. This solves problems when executing the build from a parent pom. Thanks!

davidmoten commented:
>I also added another commit which sets the working directory of the process to that of the executing Maven module. This solves problems when executing the build from a parent pom. Thanks!

Beaut, thank you.

I realize I forgot something really obvious in my review. Can you add a unit test to jax-maven-plugin-test module?

johnpoth commented:
@davidmoten added test, thanks!

davidmoten commented on jax-maven-plugin-test/pom.xml:
```diff
@@ -56,6 +56,31 @@
+                                <echo>Copying code for the schemagen with source option test</echo>
+                                <mkdir dir="${project.build.directory}/schema-src"/>
+                                <copy todir="${project.build.directory}/schema-src">
+                                    <fileset dir="${basedir}/../jax-maven-plugin-test-jaxb/src/main/java">
```
Why not reference this path directly in the source element?

johnpoth replied:
@davidmoten good point :) updated

davidmoten commented:
Release 0.1.6 to Maven Central. Normally available for download within 10 minutes (not  in the index for browsing for 24 hours or so).

johnpoth commented:
Thanks @davidmoten ! Release works like a charm
